pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -291,6 +291,60 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return measure.probs(obs_wires);
     }
 
+    /**
+     * @brief Probabilities with shot-noise for a subset of the full system.
+     *
+     * @param num_shots Number of shots.
+     * @param wires Wires will restrict probabilities to a subset
+     * of the full system.
+     *
+     * @return Floating point std::vector with probabilities.
+     */
+    auto probs(const std::vector<size_t> &wires, const size_t &num_shots)
+        -> std::vector<PrecisionT> {
+        auto counts_map = counts(num_shots);
+
+        size_t num_wires = _statevector.getTotalNumQubits();
+
+        std::vector<PrecisionT> prob_shots(size_t{1} << wires.size(), 0.0);
+
+        for (auto &it : counts_map) {
+            size_t bitVal = 0;
+            for (size_t bit = 0; bit < wires.size(); bit++) {
+                bitVal += ((it.first >> (num_wires - size_t{1} - wires[bit])) &
+                           size_t{1})
+                          << (wires.size() - size_t{1} - bit);
```

Comment:
This formula is a bit hard to follow, might benefit from a comment?

Reply:
Thanks, a comment added!